frontend/cypress/integration/common/hooks.ts

```diff
@@ -169,45 +169,48 @@ metadata:
-      }
-
-      tries++;
-      cy.request({ method: 'GET', url: '/api/mesh/graph' }).then(response => {
-        expect(response.status).to.equal(200);
-        console.log(response.body.elements.nodes.find(node => node.data.infraType === 'istiod'));
-        if (
-          response.body.elements.nodes.find(node => node.data.infraType === 'istiod')?.data?.infraData?.config
-            ?.sharedConfig === undefined
-        ) {
-          cy.log(`Kiali has not seen the Shared Mesh Config yet. Tries: ${tries}. Waiting 3s...`);
-          cy.wait(3000);
-          doRequest();
-        } else {
-          cy.request({ method: 'GET', url: '/api/namespaces/istio-system/controlplanes/istiod/metrics' }).then(
-            metricsResponse => {
-              expect(metricsResponse.status).to.equal(200);
-              console.log(metricsResponse.body);
-              if (metricsResponse.body.process_resident_memory_bytes == null) {
-                cy.log(`Istiod hasn't load the Memory metrics yet. Tries: ${tries}. Waiting 3s...`);
-                console.log(metricsResponse.body);
-                cy.wait(3000);
-                doRequest();
-              }
-            }
-          );
+  cy.exec(`kubectl get pods -l app=istiod -n istio-system -o jsonpath="{.items[0].metadata.name}"`).then(result => {
+    const podName = result.stdout;
+
+    cy.exec(`echo "${istioSharedMeshConfigMap}" | kubectl apply -f -`);
+    const patch = '{"spec": {"values": {"pilot": {"env": {"SHARED_MESH_CONFIG": "istio-user"}}}}}';
+    cy.exec(`kubectl patch istio default --type='merge' -p '${patch}'`).then(() => {
+      const maxTries = 15;
+      let tries = 0;
+      const doRequest = (): void => {
+        if (tries > maxTries) {
+          throw new Error('Timed out waiting for Kiali to see the Shared Mesh Config');
         }
-        cy.wait(5000);
-      });
-    };
-    doRequest();
+
+        tries++;
+        cy.request({ method: 'GET', url: '/api/mesh/graph' }).then(response => {
+          expect(response.status).to.equal(200);
+          console.log(response.body.elements.nodes.find(node => node.data.infraType === 'istiod'));
+          if (
+            response.body.elements.nodes.find(node => node.data.infraType === 'istiod')?.data?.infraData?.config
+              ?.sharedConfig === undefined
+          ) {
+            cy.log(`Kiali has not seen the Shared Mesh Config yet. Tries: ${tries}. Waiting 3s...`);
+            cy.wait(3000);
+            doRequest();
+          } else {
+            const waitForPodDeletion = (): void => {
+              cy.exec(`kubectl get pod ${podName} -n istio-system`, { failOnNonZeroExit: false }).then(res => {
+                if (res.code === 0) {
+                  cy.wait(2000).then(waitForPodDeletion);
+                } else {
+                  cy.log(`Pod ${podName} is deleted.`);
+                }
+              });
+            };
+
+            waitForPodDeletion();
+          }
```

Comment:
You can wait for the pod to be deleted with `kubectl wait` after patching istio: `kubectl wait --for=delete pod/busybox1 --timeout=60s`